src/test/java/spoon/test/generics/GenericsTest.java

```diff
@@ -695,36 +725,39 @@ public void testCtSuperTypeHierarchy() throws Exception {
 		
 		//type of first parameter of  method WeddingLunch#eatMe
 		CtTypeReference<?> ctWeddingLunch_X = tWeddingLunch_eatMe.getParameters().get(0).getType();
+		// X is the type parameter of WeddingLunch
 		assertEquals("X", ctWeddingLunch_X.getSimpleName());
 		//type of first parameter of method Lunch#eatMe
 		CtTypeReference<?> ctClassLunch_A = ctClassLunch_eatMe.getParameters().get(0).getType();
 		assertEquals("A", ctClassLunch_A.getSimpleName());
 		
 		//are these two types same?
-		CtSuperTypeHierarchy sthOftrWeddingLunch_Mole = new CtSuperTypeHierarchy(trWeddingLunch_Mole);
+		CtSuperTypeHierarchy typingContextOfDisgust = new CtSuperTypeHierarchy(typeReferenceOfDisgust);
 		
-		//adapt X to scope of CelebrationLunch<Integer,Long,Double>.WeddingLunch<Mole>
-		CtTypeReference<?> adapted = sthOftrWeddingLunch_Mole.adaptType(ctWeddingLunch_X);
-		assertEquals("spoon.test.generics.testclasses.Mole", adapted.getQualifiedName());
+		// in disgust, X of WeddingLunch is bound to "Model"
+		assertEquals("spoon.test.generics.testclasses.Mole", typingContextOfDisgust.adaptType(ctWeddingLunch_X).getQualifiedName());
 		//adapt A to scope of CelebrationLunch<Integer,Long,Double>.WeddingLunch<Mole>
-		adapted = sthOftrWeddingLunch_Mole.adaptType(ctClassLunch_A);
-		assertEquals("spoon.test.generics.testclasses.Mole", adapted.getQualifiedName());
-		//adapt A to scope of enclosing class of CelebrationLunch<Integer,Long,Double>.WeddingLunch<Mole>, which is CelebrationLunch<Integer,Long,Double>
-		adapted = sthOftrWeddingLunch_Mole.getEnclosingGenericTypeAdapter().adaptType(ctClassLunch_A);
-		assertEquals("java.lang.Double", adapted.getQualifiedName());
-		
+
+		// in disgust, the A of Lunch is bound to "Mole"
+		assertEquals("spoon.test.generics.testclasses.Mole", typingContextOfDisgust.adaptType(ctClassLunch_A).getQualifiedName());
+
+		// I don't understand the goal and utility of this one
+		assertEquals("java.lang.Double", typingContextOfDisgust.getEnclosingGenericTypeAdapter().adaptType(ctClassLunch_A).getQualifiedName());
```

Comment:
I am sorry, "parent" was not the best name. I mean "declaring type". See example:
```java
class A1<K1> {
  class B1<L1> {
  <M1> M1 method(K1 k, L1 l){}
  }
}
class A2<K2> extends A1<K2> {
  class B2<L2> extends B1<L2> {
  <M2> M2 method(K2 k, L2 l){}
  }
}
```
In the model above the 
```java
//adapter for method `A2.B2#method`
CtMethodSuperTypeHierarchy methodAdapter = new CtMethodSuperTypeHierarchy(...method("A2.B2#method"));

//the enclosing adapter of `methodAdapter` is adapter of inner class `A2.B2`
CtSuperTypeHierarchy b2Adapter = methodAdapter.getEnclosingGenericTypeAdapter();
//the enclosing adapter of `b2Adapter ` is adapter of top level class `A2`
CtSuperTypeHierarchy a2Adapter = b2Adapter.getEnclosingGenericTypeAdapter();
//the enclosing adapter of `a2Adapter ` is null, because A2 is top level class
assertNull(a2Adapter.getEnclosingGenericTypeAdapter())
```
Note that `methodAdapter` internally needs adapters of all enclosing declaring types recursively, because that method can contain typed parameters of 
1) method itself - there is type parameter M2
2) declaring type B2 - there is type parameter L2
3) declaring type A2 - there is type parameter K2